internal/operator-controller/applier/helm.go

```diff
@@ -164,6 +197,34 @@ func (h *Helm) buildHelmChart(bundleFS fs.FS, ext *ocv1.ClusterExtension) (*char
 	return h.BundleToHelmChartFn(bundleFS, ext.Spec.Namespace, watchNamespace)
 }
 
+func (h *Helm) template(ctx context.Context, ext *ocv1.ClusterExtension, chrt *chart.Chart, values chartutil.Values, post postrender.PostRenderer) (*release.Release, error) {
```

Comment:
would it make sense to merge it with `getReleaseState` somehow? I guess the main thing is i.ClientOnly and the fact that we need to instantiate a separate client for that.
If it doesn't and ends up clumsy, I'd still probably adjust the name to be closer to`getReleaseState`, since logically it's almost the same